### ISSUE TITLE
Deps.json file not rebuild when reference library remove

### ISSUE THREAD

**Issue by manojsitapara** [dotnet/msbuild#9168]
### Issue Description

Why is my deps.json file not being rebuilt after removing a reference library in MSBuild? Application runs into 'Missing reference Excel.dll' error even though the exe file is rebuilt successfully. What steps are required to trigger the regeneration of the deps.json file when a reference library is removed?"

### Steps to Reproduce

Remove reference library from project
Build solution using MSBuild

### Expected Behavior

.exe file & deps.json should remove the reference of the library which was removed

### Actual Behavior

.exe file rebuild but deps.json is not rebuild and it has the reference of the library which was removed

**Comment by AR-May** [comment]
> What steps are required to trigger the regeneration of the deps.json file when a reference library is removed?

You could try to rebuild the solution, it should regenerate the file. But that should not be required.

Closing this issue since it is a duplicate of https://github.com/dotnet/sdk/issues/23935

**Comment by geetmenon** [comment]
The above comment by @AR-May :
> You could try to rebuild the solution, it should regenerate the file. But that should not be required.

is not a resolution for the issue we are facing with out automated build. Please provide a proper fix/resolution that we include in our automated MSBuild script(s).

**Comment by rainersigwald** [comment]
@geetmenon I see you made that same comment on the canonical bug--what are you hoping to achieve making it on this duplicate?

**Comment by geetmenon** [comment]
I have been waiting on a resolution from Microsoft on this issue for more than 18 months now. And to see a response like the the above ("You could try to rebuild the solution...") from @AR-May is so disappointing.  I added the above comment therefore in the hope to make of you aware of the futility of such irresponsible comment, and of the urgency of the situation so that users facing this issue will receive a proper fix (at the earliest) that we can incorporate into our Automated MSBuild scripts so that we are tired of explaining to our Dev Mgrs and CTO each time as to why we are having to take manual steps during our build & deployment process (despite having a comprehensive automated build scripts for all our solutions) because Microsoft has still not provided a solution to fix this issue even after more than 18 months of us reporting the issue.
